File: count_sort.py
d0 = [0,1,3,2,4,5,3,8,4,5,6]
maxnum = max(d0)
length = len(d0)
C = [0 for i in range(maxnum+1)]
B = [0 for i in range(length)]
for i in range(length):
    C[d0[i]]+=1
for i in range(1,maxnum+1):
    C[i] = C[i]+C[i-1]
for i in range(length):
    B[C[d0[i]]-1] = d0[i]
    C[d0[i]]-=1
print(B)

File: Sort/countsort.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> countsort(vector<int> num){
    int max_num = 0;
    for(auto i : num){
        max_num =  max(max_num, i);
    }
    vector<int> rec(max_num+1, 0);
    vector<int> ans;
    for(auto i : num){
        rec[i] += 1;
    }
    for(int i = 0;i <= max_num;++i){
        if(rec[i]){
            for(int j = 0;j < rec[i];++j){
                ans.push_back(i);
            }
        }
    }
    return ans;
}
int main(){
    vector<int> num {1,2,6,4,3,52,21,54,32,345};
    vector<int> ans = countsort(num);
    for(auto i : ans){
        cout<<i<<"  ";
    }
    cout<<endl;
}